### DISCUSSION TITLE
[RFC] A deployment builder should have an interactive prompt after the "ng add".

### DISCUSSION THREAD

**Discussion by JohannesHoppe** [angular-schule/ngx-deploy-starter#2] [Ideas]
To make it easier for the end user to get started, a deployment builder **should** ask for all the mandatory questions immediately after the `ng add`.
The data should be persisted in the `angular.json` file.

This is about a shared experience across the builders. I suggest

1. an interactive  prompt (e.g. via [schematic-prompts](https://github.com/angular/angular-cli/blob/master/docs/specifications/schematic-prompts.md))
2. that can be disabled with an option like `--defaults` (similar to `ng new X --defaults`).

**Comment by fmalcher** [comment]
Agree! All prompted options should be possible to set via command-line options as well.
I like the way `ng new` does this. You can supply the `--routing` option, but if you don't it will ask you (because it's likely you just forgot it).

tl;dr;

- Yes, prompt after `ng add` should be there
- there must be a way to run fully promptless
- all prompted options must be available as cmd line options as well
- but: not all possible options need to be prompted

**Comment by shmool** [comment]
I agree about prompting the options, and this is how I implemented it in [@Azure/ng-deploy](https://github.com/Azure/ng-deploy-azure/).

Questions:

1. The default is a quickstart, which asks the minimal amount of questions and creates the necessary resources (i.e. storage account) with arbitrary names and values. There's a `manual` flag which gives all the prompts. You can also pass configuration (flags) with the resources names (new or existing resources) - whatever is not passed will be given arbitrarily.
What do you think? Should the manual mode be the default instead?

2. We save the configuration (Azure resources) in `azure.json`. We saw similar examples in other projects. Is that ok, or should we save it in `angular.json`?

**Comment by JohannesHoppe** [comment]
@shmool Many thanks for your feedback.

1. That sounds very reasonable to me. I suggest to use as many defaults as possible and only ask the questions that are absolutely necessary.

2. I think the `angular.json` is the right place for the configurations. In the past, a standard CLI project consisted of more configuration files. Many of them have been removed, which has increased usability and hided complexity. I will now create a separate RFC for this.
